Project1/CellEdge.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+class GridCell;
+#include <array>
+#include "MobilityList.h"
+/**
+Represents a Path through some tiles, basically a struct
+*/
+class CellEdge {
+	MobilityList _costs;
+	MobilityList _traversable;
```

Comment:
Why would this need to be a MobilityList? The edge is either going to have a cost for a certain mobility type, in which case it's traversal by that type, or it won't have a cost for that type and therefore is not.

I think what you really need is an additional mobility type that is for attacks only. PROJECTILE maybe.